specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1178 @@
+order in the result xml will be the order it was added to the builder.
+
+# AppNotificationComboBox
+
+AppNotificationComboBox is an `<input>` component that allows users to pick from a dropdown menu of
+values on an AppNotification.
+
+**WinAppSDK 1.2 AppNotificationComboBox Schema**:
+
+```xml
+<input id = string
+        type="selection"
+        defaultInput? = string>
+    <!-- Child elements -->
+    selection{0,5}
+</input>
+```
+
+These attributes are abstracted with the AppNotificationComboBox component APIs:
+
+-   AppNotificationComboBox(String id)
+-   SetDefaultSelection(String id)
+-   AddSelection(String id, String content)
+
+Below is an example use:
+
+![ComboBox Example](ComboBoxExample.png)
+
+```cpp
+AppNotificationContent()
+    .AddComboBox(AppNotificationComboBox(L"ComboBox")
+        .SetDefaultSelection(L"yes")
+        .AddSelection(L"yes", L"Going")
+        .AddSelection(L"maybe", L"Maybe")
+        .AddSelection(L"no", L"Decline"))
+    .AddButton(AppNotificationButton(L"Send")
+        .AddArgument(L"action", L"Send"))
+        .SetInputId(L"ComboBox")
+    .GetXml();
+```
+
+XML output:
+
+```xml
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+    </binding>
+  </visual>
+
+  <actions>
+    <input id="ComboBox" type="selection" defaultInput="yes">
+      <selection id="yes" content="Going" />
+      <selection id="maybe" content="Maybe" />
+      <selection id="no" content="Decline" />
+    </input>
+    <action
+      arguments="action=Send"
+      hint-inputId="ComboBox"
+      content="Send"/>
+
+  </actions>
+</toast>
+```
+
+# AppNotificationProgressBar
+
+AppNotificationProgressBar is a component that builds the `<progress>` xml.
+AppNotificationProgressBar is used to update an AppNotification that is currently displayed. This
+allows developers to inform users about the status of certain operations such as download or
+install.
+
+**WinAppSDK 1.2 `<progress>` Schema**:
+
+```xml
+<progress title? = string
+    status = string
+    value = string
+    valueStringOverride? = string
+/>
+```
+
+AppNotificationProgressBar uses data binding to update the active UI. In the WindowsAppSDK, this is
+done by using AppNotificationProgressData. Click
+[Here](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/AppNotifications/AppNotifications-spec.md#notification-progress-updates)
+for an example.
+
+All the attributes have static data binding constants:
+
+-   Status = "{progressStatus}"
+-   Value = "{progressValue}"
+-   Title = "{progressTitle}"
+-   ValueStringOverride = "{progressValueString}
+
+Every AppNotificationProgressBar component will have the bound status and value. Title and
+ValueStringOverride are not required so those constants will not be added unless you call
+the AppNotificationProgressBar component APIs
+
+-   UseTitleForLabel(String title)
+-   BindTitleForLabel()
+-   BindValueStringOverride()
+
```

Comment:
Don't we also need
* `SetStatus` / `BindStatus`
* `SetValue` / `BindValue`
* `SetValueStringOverride` / `BindValueStringOverride`